proto/site/v1alpha/site.proto

```diff
@@ -0,0 +1,207 @@
+syntax = "proto3";
+
+package com.mintter.site.v1alpha;
+
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/empty.proto";
+
+option go_package = "mintter/backend/genproto/site/v1alpha;site";
+
+// Site API service allows to configure a remote Mintter Site.
+// The server exposing this API should take care about authenticating the caller.
+service Site {
```

Comment:
This is the actual remote Site API. It can be exposed by the site itself over libp2p (so that it's authenticated). And it can also be exposed by the local backend (daemon) as any other existing API. The daemon should require a header to specify the hostname to talk to.

Reply:
Ok, so we need to make sure not to expose that API on the regular gRPC API as well (in the remote p2p server). Currently, we don't make that distinction with any gRPC endpoint, right?. Aall p2p reachable APIs like the remote invoice is locally reachable as well). Is that something we can easily do?

Reply:
I'm not sure I understand the question, but let me explain a bit more how it's currently set up and how can we leverage that.

- Inside the daemon we have 2 `[grpc.Server](https://pkg.go.dev/google.golang.org/grpc#Server)` instances. You have to bind the server to a `net.Listener` to expose it.
  - One is bound on the plain TCP port and is used by the frontend client.
  - The second one is bound to the libp2p stream using the [gostream](https://github.com/libp2p/go-libp2p-gostream) package. It uses the libp2p stream as a `net.Listener`.
 - Each of these 2 servers can have separate sets of GRPC API interfaces. The registration is performed by calling the corresponding method in the autogenerated code, passing the `grpc.Server` instance.

Currently everything except the `p2p` proto is exposed on the TCP listener, and the P2P stuff is exposed on the libp2p listener.

With this new Site API we would expose it on the TCP listener locally - so the frontend client can call the local daemon and we would proxy to the remote site.

And we would also expose this Site API on the remote site over libp2p, so that our local daemon can actually proxy the request.

Does this answer your question?

Reply:
Actually, I don't know if all of this is explained somewhere in the code, but if not, may be worth putting it somewhere.

Reply:
Yes, it helps so much!